R/lasso-development.R

```diff
@@ -165,181 +165,261 @@
       # Predictions (in terms of probability)
       private$predictions <- predict(object = private$fitGrLasso,
-                                    X = model.matrix(private$modFmla, data = private$dfTestTemp)[,-1],
-                                    lambda = private$lambda1se,
-                                    type = "response")
+                                     X = model.matrix(private$modFmla, data = private$dfTestTemp)[,-1],
+                                     lambda = private$lambda1se,
+                                     type = "response")
       
       if (isTRUE(self$params$debug)) {
         print(paste0("Rows in prob prediction: ", nrow(private$predictedVals)))
-        print("First 10 raw classification probability predictions")
-        print(round(private$predictions[1:10], 2))
+        if (self$params$type == 'classification') {
+          print("First 10 raw classification probability predictions")
+          print(round(private$predictions[1:10], 2))
+        } else if (self$params$type == 'regression') {
+          print("First 10 raw regression value predictions")
+          print(round(private$predictions[1:10], 2))
+        }
       }
     },
-
+    
     # Generate performance metrics
     generatePerformanceMetrics = function() {
-
       ytest <- as.numeric(private$dfTest[[self$params$predictedCol]])
       
-      calcObjList <- calculatePerformance(private$predictions, 
-                                           ytest, 
-                                           self$params$type)
+      calcObjList <-
+        calculatePerformance(private$predictions,
+                             ytest,
+                             self$params$type)
       
       # Make these objects available for plotting and unit tests
       private$ROCPlot <- calcObjList[[1]]
-      private$PRCurvePlot <- calcObjList[[2]]
+      private$PRCurvePlot <-
+        calcObjList[[2]]
       private$AUROC <- calcObjList[[3]]
       private$AUPR <- calcObjList[[4]]
       private$RMSE <- calcObjList[[5]]
       private$MAE <- calcObjList[[6]]
-
+      
       if (isTRUE(self$params$printResults)) {
         print("Grouped Lasso coefficients:")
-        print(private$fitGrLasso$fit$beta[,private$indLambda1se])
+        print(private$fitGrLasso$fit$beta[, private$indLambda1se])
       }
-
+      
       if (isTRUE(self$params$varImp)) {
-        imp = names(private$dfTrainTemp[ ,!(colnames(private$dfTrainTemp) == self$params$predictedCol)])[predict(private$fitGrLasso, private$modMat,type = "groups", lambda = private$lambda1se)]
-        print(paste0("Variables with non-zero coefficients: ",paste0(imp, collapse = ", ")))
+        imp = names(private$dfTrainTemp[, !(colnames(private$dfTrainTemp) == self$params$predictedCol)])[predict(
+          private$fitGrLasso,
+          private$modMat,
+          type = "groups",
+          lambda = private$lambda1se
+        )]
+        print(paste0(
+          "Variables with non-zero coefficients: ",
```

Comment:
`"Variables` can go up on the previous line, especially after changing `print` to `cat` (since `cat` will let you get rid of the first `paste0`.